src/gui/editors/Editor.cpp

```diff
@@ -104,36 +104,40 @@ Editor::Editor(bool record, bool stepRecord) :
 		m_toolBar->widgetForAction(action)->setObjectName(objectName);
 	};
 
-	// Set up play and record actions
+	auto createConditionalButton = [this, &addButton](
+		bool condition,
+		QAction*& targetAction,
+		const char* iconName,
+		const QString& text,
+		const char* slot,
+		const QString& objectName
+	) {
+		if (condition) {
+			targetAction = new QAction(embed::getIconPixmap(iconName), text, this);
```

Comment:
Would this be a good opportunity to change the `QAction` pointers to `unique_ptr`?